zaza/charm_lifecycle/utils.py

```diff
@@ -22,6 +23,137 @@
 
 BUNDLE_DIR = "./tests/bundles/"
 DEFAULT_TEST_CONFIG = "./tests/tests.yaml"
+DEFAULT_MODEL_ALIAS = "default_alias"
+
+
+def _model_alias_str_fmt(data):
+    """Convert to model_alias:str format if needed.
+
+    If a string is passed in then the string is mapped to the
+    DEFAULT_MODEL_ALIAS. If a dict is passed then this is a noop.
+
+    :param data: String or Model Alias to data map
+    :type data: str or {}
+    :returns: Model Alias to data map
+    :rtype: {}
+    """
+    if isinstance(data, collections.Mapping):
+        return data
+    else:
+        return {DEFAULT_MODEL_ALIAS: data}
+
+
+def _concat_model_alias_maps(data):
+    """Iterate over list and construct single dict of model alias maps.
+
+    Any elements in list which are not dicts are added to a list and assigned
+    to DEFAULT_MODEL_ALIAS.
+
+    eg If input is ['e1', 'e2', {'alias1': ['e3'], 'alias2': ['e4']}]
+       this function will return:
+       {
+           DEFAULT_MODEL_ALIAS: ['e1', 'e2'],
+           'alias1': ['e3'],
+           'alias2': ['e4']}
+
+    :param data: List comprised of str elements or dict elements.
+    :type data: []
```

Comment:
type data: List[???]

It looks a bit like:

`List[Union[str, Dict[str, List[str]]]`

from the docstring comment.